addons/web/static/tests/views/form/form_view.test.js

```diff
@@ -1165,6 +1166,63 @@ test(`invisible fields are properly hidden`, async () => {
     expect(`.o_field_widget[name="child_ids"]`).toHaveCount(0);
 });
 
+test(`invisible fields are properly hidden (company evalContext)`, async () => {
+    Partner._records[0].int_field = 3;
+    serverState.companies = [
+        {
+            id: 1,
+            name: "Company 1",
+            sequence: 1,
+            parent_id: false,
+            child_ids: [],
+            country_code: "BE",
+        },
+        {
+            id: 2,
+            name: "Company 2",
+            sequence: 2,
+            parent_id: false,
+            child_ids: [],
+            country_code: "PE",
+        },
+        {
+            id: 3,
+            name: "Company 3",
+            sequence: 3,
+            parent_id: false,
+            child_ids: [],
+            country_code: "AR",
+        },
+    ];
+    cookie.set("cids", "3-1");
+
+    await mountView({
+        resModel: "partner",
+        type: "form",
+        // the arch contains an x2many field without inline view: as it is always invisible,
+        // the view should not be fetched. we don't specify any view in this test, so if it
+        // ever tries to fetch it, it will crash, indicating that this is wrong.
+        arch: `
+            <form>
+                <sheet>
+                    <field name="int_field" invisible="1"/>
+                    <group>
+                        <field name="foo" invisible="not companies.has(companies.active_company_ids, 'country_code', 'PE')"/>
+                        <field name="bar"/>
+                    </group>
+                    <field name="float_field" invisible="not companies.has(companies.current_company_id, 'country_code', 'PE')"/>
+                    <field name="child_ids" invisible="not companies.has(int_field, 'country_code', 'PE')"/>
```

Comment:
Using the prefix `companies.X` totally makes sense, but what's the point of repeating `company` within this namespace?
* `companies.active_company_ids` → `companies.active_ids`
* `companies.current_company_id` → `companies.current_id`

Moreover, the use of `active` vs `current` seems to reproduce the web client's internal idiosyncrasies.  When performing some action on a selection of records in a list view, the context keys to retrieve the first selected record and all selected records are `active_id` and `active_ids`, respectively.  Why don't we use the same convention here?